dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -0,0 +1,752 @@
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.tracing.RESTEasyTracingLogger;
+import org.jboss.resteasy.util.CommitHeaderAsyncOutputStream;
+import org.jboss.resteasy.util.CommitHeaderOutputStream;
+import org.jboss.resteasy.util.CommitHeaderOutputStream.CommitCallback;
+import org.jboss.resteasy.util.HttpHeaderNames;
+import org.jboss.resteasy.util.MediaTypeHelper;
+
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.WriterInterceptor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+public class ServerResponseWriter
+{
+ // Liberty change begin
+    // caches for static MediaTypes from the API - saves the cost of building a new instance for every response 
+    private static final Map<String, MediaType> mediaTypeCache;
+    private static final Map<MediaType, MediaType> mediaTypeCharsetCache;
+    static {
+        Map<String, MediaType> mtCache = new HashMap<>();
+        mtCache.put(MediaType.APPLICATION_ATOM_XML, MediaType.APPLICATION_ATOM_XML_TYPE);
+        mtCache.put(MediaType.APPLICATION_FORM_URLENCODED, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
+        mtCache.put(MediaType.APPLICATION_JSON_PATCH_JSON, MediaType.APPLICATION_JSON_PATCH_JSON_TYPE);
+        mtCache.put(MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON_TYPE); 
+        mtCache.put(MediaType.APPLICATION_OCTET_STREAM, MediaType.APPLICATION_OCTET_STREAM_TYPE);
+        mtCache.put(MediaType.APPLICATION_SVG_XML, MediaType.APPLICATION_SVG_XML_TYPE);
+        mtCache.put(MediaType.APPLICATION_XHTML_XML, MediaType.APPLICATION_XHTML_XML_TYPE);
+        mtCache.put(MediaType.APPLICATION_XML, MediaType.APPLICATION_XML_TYPE);
+        mtCache.put(MediaType.MULTIPART_FORM_DATA, MediaType.MULTIPART_FORM_DATA_TYPE);
+        mtCache.put(MediaType.SERVER_SENT_EVENTS, MediaType.SERVER_SENT_EVENTS_TYPE);
+        mtCache.put(MediaType.TEXT_HTML, MediaType.TEXT_HTML_TYPE);
+        mtCache.put(MediaType.TEXT_PLAIN, MediaType.TEXT_PLAIN_TYPE);
+        mtCache.put(MediaType.TEXT_XML, MediaType.TEXT_XML_TYPE);
+        mtCache.put(MediaType.WILDCARD, MediaType.WILDCARD_TYPE);
```

Comment:
```suggestion
```

WILDCARD is not concrete so the way things are coded right now, this entry is not used.

Reply:
Good catch - removed WILDCARD